### PR TITLE
[#534] Add specs for MoodsController and CategoriesController

### PR DESCRIPTION
**Pull request by timorthi**
Summary of changes:
* Added spec for MoodsController
* Removed deprecated logic in MoodsController#show
* Added spec for CategoriesController
* Removed deprecated logic in CategoriesController#show

### CODE REVIEW

**Comment by timorthi** [comment]
@HashNotAdam Both specs are ready for review now :)

File: spec/controllers/categories_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+RSpec.describe CategoriesController, type: :controller do
+  let(:user) { create(:user1) }
+  let(:category) { create(:category, userid: user.id) }
+  let(:other_category) { create(:category, userid: user.id + 1) }
+  let(:valid_category_params) { attributes_for(:category).merge(userid: user.id) }
+  let(:invalid_category_params) { { name: nil, description: nil } }
+
+  describe 'GET #index' do
+    context 'when the user is logged in' do
```
Comment (HashNotAdam) [inline spec/controllers/categories_controller_spec.rb]: Given the meta-programming involved in setting @categories, it's probably worth verifying it is set correctly and that @page_tooltip is set.

File: spec/controllers/moods_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+RSpec.describe MoodsController, type: :controller do
+  let(:user) { create(:user1) }
+  let(:user_mood) { create(:mood, userid: user.id) }
+  let(:other_mood) { create(:mood, userid: user.id + 1) }
+  let(:valid_mood_params) { attributes_for(:mood).merge(userid: user.id) }
+  let(:invalid_mood_params) { { name: nil, description: nil } }
+
+  describe 'GET #index' do
+    context 'when the user is logged in' do
```
Comment (HashNotAdam) [inline spec/controllers/moods_controller_spec.rb]: Given the meta-programming involved in setting @moods, it's probably worth verifying it is set correctly and that @page_tooltip is set.

**Comment by HashNotAdam** [comment]
Nice work, @timorthi 
I have to hold myself back from completely refactoring those controllers now that there are tests 😆 

We'll ignore the Code Climate issues on this one since they are related to the controllers themselves.

File: spec/controllers/categories_controller_spec.rb (hunk 1 above)
Comment (timorthi) [inline spec/controllers/categories_controller_spec.rb]: @HashNotAdam Done!

Comment (HashNotAdam) [inline spec/controllers/categories_controller_spec.rb]: Thanks, this doesn't check that categories is set correctly, though.
`expect(assigns(:categories)).to eq [category]`

page_tooltip 👍

**Comment by julianguyen** [comment]
Besides the Codeclimate, is this ready to merge? :)

**Comment by HashNotAdam** [comment]
@julianguyen I've requested a change so that the test on index is actually checking that the instance variable is set correctly.
Also, the branch needs to be rebased.

**Comment by timorthi** [comment]
@HashNotAdam Fixed and rebased :)

**Comment by HashNotAdam** [comment]
LGTM!

**Pull request merged**